### PR TITLE
feat(input): fill and outline support

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Developers could not specify the shape or the fill on `ion-input` with the new syntax.

## What is the new behavior?

- Deprecated `fill` and `shape` on `ion-item`.
- Added template examples for outline/shape.
- Updated a11y tests to ensure new styles pass Axe criteria.
- Added screenshot tests for new styles.
- Added styles for rounded shape.
- Added styles for fill="solid".
- Added styles for fill="outline". Most of these styles are based off https://material-components.github.io/material-components-web-catalog/#/component/text-field.
- Updated the bottom content to take on the start/end padding so it always lines up with the label/text input.

Note: I discovered an issue with all inputs where clicking the far edges of the container (left and right) does not activate the input. It is more easily reproduced with the shape/fill inputs than the default inputs. I plan on addressing the issue in all of the inputs in a separate PR.

This PR does not include the highlight functionality (i.e. the border line appearing red when invalid, green when valid etc). This will be done in a separate PR.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/input/input.scss [hunk 1]
```diff
@@ -219,23 +233,33 @@
...
+  position: relative;
+
   flex-grow: 1;
 
   align-items: stretch;
 
   height: 100%;
 
-  background: inherit;
+  transition: background-color 15ms linear;
```
Comment (amandaejohnston) [inline core/src/components/input/input.scss:248]: Is this transition meant to apply to the `solid` style? If so, I'm not sure if it's working; I tried changing it to `1000ms` but I'm not able to see the animation.

File: core/src/components/input/test/fill/input.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          label="Email"
+          value="hi@ionic.io"
+          helper-text="Enter your email"
+          maxlength="20"
+          counter="true"
+        ></ion-input>
+      `);
+
+      const input = page.locator('ion-input');
+      expect(await input.screenshot()).toMatchSnapshot(`input-fill-shaped-outline-${page.getSnapshotSettings()}.png`);
```
Comment (amandaejohnston) [inline core/src/components/input/test/fill/input.e2e.ts:72]: The screenshots here look the same as the non-rounded version for some reason? Ditto for the `solid` versions, actually; seems like the `rounded` look isn't being rendered properly.

File: core/src/components/input/input.scss (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components/input/input.scss:248]: How are you testing this? It's working on my end.

File: core/src/components/input/test/fill/input.e2e.ts (hunk 2 above)
Comment (liamdebeasi) [reply core/src/components/input/test/fill/input.e2e.ts:72]: Whoops the keyword is `round` not `rounded`. Just updates + re-ran the screenshot job.

File: core/src/components/input/input.scss (hunk 1 above)
Comment (amandaejohnston) [reply core/src/components/input/input.scss:249]: Just hovering/focusing the inputs in the `fill` test: https://user-images.githubusercontent.com/90629384/200656142-3c8715f1-f21f-48b5-a027-abe1c85cbdae.mp4

Comment (liamdebeasi) [reply core/src/components/input/input.scss:249]: Not really sure. It's working on my end (I changed the duration to 300ms for emphasis):

https://user-images.githubusercontent.com/2721089/200656831-3872843d-50f1-461b-a51c-2c6d217454e8.mov

Comment (amandaejohnston) [reply core/src/components/input/input.scss:249]: I'm on Windows Chrome if it helps 🤔

Comment (liamdebeasi) [reply core/src/components/input/input.scss:249]: Does the animation work here? https://codepen.io/liamdebeasi/pen/VwdPYKw

Comment (amandaejohnston) [reply core/src/components/input/input.scss:249]: That's working on my end, yeah.

Comment (amandaejohnston) [reply core/src/components/input/input.scss:249]: We chatted on Slack and it seems to be something with my setup 🤷 I'm cool moving forward.

**Review by amandaejohnston — APPROVED** [review]

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**